### PR TITLE
feat: Clone identities (FE)

### LINKED ISSUE #2871: Compare / Clone Identities
Add the ability in the dashboard to compare flags between two Identities and then copy/clone the flags from Identity A to B

### PR DESCRIPTION
**Pull request by novakzaballa**
Thanks for submitting a PR! Please check the boxes below:

- [x] I have run [`pre-commit`](https://docs.flagsmith.com/platform/contributing#pre-commit) to check linting
- [x] I have added information to `docs/` if required so people know about the feature!
- [x] I have filled in the "Changes" section below?
- [x] I have filled in the "How did you test this code" section below?
- [x] I have used a [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) title for this Pull Request

## Changes

_Please describe._

- Add the possibility to clone all feature states from an identity A to identity B

- Create 2 identities in an environment
- Create feature state overrides in both
- Go to Compare in the side menu -> Go to the Identities tab
- Select the identities and click in the clone button

### CODE REVIEW

**Review by Jayjogarajiya67 — COMMENTED** [review]
Nice project

File: frontend/common/types/responses.ts [hunk 1]
```diff
@@ -287,6 +287,7 @@ export type IdentityFeatureState = {
   enabled: boolean
   feature_state_value: FlagsmithValue
   segment: null
+  overridden_by: string
```
Comment (kyle-ssg) [inline frontend/common/types/responses.ts]: I'm guessing this is string|null

File: frontend/web/components/CompareIdentities.tsx [hunk 2]
```diff
@@ -120,6 +127,39 @@ const CompareIdentities: FC<CompareIdentitiesType> = ({
...
+    leftIdentityName: string,
+    rightIdentityName: string,
+    leftIdentityId: string,
+    rightIdentityId: string,
+    environmentId: string,
+  ) => {
+    return openConfirm({
+      body: (
+        <div>
+          {'Are you sure you want to clone all the feature states from '}
```
Comment (kyle-ssg) [inline frontend/web/components/CompareIdentities.tsx]: Does cloning clone overrides and traits?

If so I think that might read better

"Cloning an identity will copy any traits and identity overrides. Are you sure?"

File: frontend/web/components/CompareIdentities.tsx [hunk 3]
```diff
@@ -120,6 +127,39 @@ const CompareIdentities: FC<CompareIdentitiesType> = ({
...
+          body: {
+            source_identity_id: leftIdentityId,
+          },
+          environment_id: environmentId,
+          identity_id: rightIdentityId,
+        }).then(() => {
+          toast('Clonation Completed!')
+        })
+      },
+      title: 'Clone the values',
```
Comment (kyle-ssg) [inline frontend/web/components/CompareIdentities.tsx]: Clone identity might be better here ? Not clear what "The values" means

File: frontend/web/components/CompareIdentities.tsx (hunk 2 above)
Comment (novakzaballa) [reply frontend/web/components/CompareIdentities.tsx]: We don't clone the traits.
Maybe could be: "Cloning {identityA} will copy any identity overrides in {identityB}. Are you sure?"

File: frontend/web/components/CompareIdentities.tsx (hunk 3 above)
Comment (novakzaballa) [reply frontend/web/components/CompareIdentities.tsx]: Changed

File: frontend/common/types/responses.ts (hunk 1 above)
Comment (novakzaballa) [reply frontend/common/types/responses.ts]: Corrected

File: frontend/web/components/CompareIdentities.tsx [hunk 4]
```diff
@@ -179,9 +218,40 @@ const CompareIdentities: FC<CompareIdentitiesType> = ({
...
+                              leftId?.label,
+                              rightId?.label,
+                              leftId?.value,
+                              rightId?.value,
+                              environmentId,
+                            )
+                          }}
+                          className='ms-2 me-2'
+                        >
+                          {'Clone features states'}
```
Comment (kyle-ssg) [inline frontend/web/components/CompareIdentities.tsx]: Clone identity overrides I think?

File: frontend/web/components/CompareIdentities.tsx [hunk 5]
```diff
@@ -179,9 +218,40 @@ const CompareIdentities: FC<CompareIdentitiesType> = ({
...
+                        </Button>
+                      }
+                    >
+                      {`Clone the features states from ${leftId?.label} to ${rightId?.label}`}
```
Comment (kyle-ssg) [inline frontend/web/components/CompareIdentities.tsx]: Same here

File: frontend/web/components/CompareIdentities.tsx [hunk 6]
```diff
@@ -179,9 +218,40 @@ const CompareIdentities: FC<CompareIdentitiesType> = ({
...
+            title={'Changed Flags'}
+            className='mt-3'
+            cta={
+              <>
+                {Utils.getFlagsmithHasFeature('clone_identities') && (
+                  <>
+                    <Tooltip
+                      title={
+                        <Button
+                          onClick={() => {
```
Comment (kyle-ssg) [inline frontend/web/components/CompareIdentities.tsx:232]: disabled={!leftId||!rightId||!environmentId}

File: frontend/common/types/responses.ts [hunk 7]
```diff
@@ -466,7 +467,7 @@ export type Res = {
   rolePermission: PagedResponse<UserPermission>
   projectFlags: PagedResponse<ProjectFlag>
   projectFlag: ProjectFlag
-  identityFeatureStates: IdentityFeatureState[]
+  identityFeatureStatesAll: IdentityFeatureState[]
```
Comment (kyle-ssg) [inline frontend/common/types/responses.ts:674]: Im a bit confused, why was this renamed?

File: frontend/web/components/CompareIdentities.tsx [hunk 8]
```diff
@@ -120,6 +127,38 @@ const CompareIdentities: FC<CompareIdentitiesType> = ({
     )
   }
 
+  const clonedentityValues = (
```
Comment (kyle-ssg) [inline frontend/web/components/CompareIdentities.tsx]: cloneIdentityValues

File: frontend/web/components/CompareIdentities.tsx (hunk 4 above)
Comment (novakzaballa) [reply frontend/web/components/CompareIdentities.tsx]: This text was updated.

File: frontend/web/components/CompareIdentities.tsx (hunk 5 above)
Comment (novakzaballa) [reply frontend/web/components/CompareIdentities.tsx]: this text was updated.

File: frontend/web/components/CompareIdentities.tsx (hunk 6 above)
Comment (novakzaballa) [reply frontend/web/components/CompareIdentities.tsx:232]: Good catch, Added

File: frontend/web/components/CompareIdentities.tsx (hunk 8 above)
Comment (novakzaballa) [reply frontend/web/components/CompareIdentities.tsx]: Good catch, was corrected

File: frontend/common/types/responses.ts (hunk 7 above)
Comment (novakzaballa) [reply frontend/common/types/responses.ts:674]: I think this name is clearer to refer to the endpoints: `/api/v1/environments/{environment_api_key}/edge-identities/{edge_identity_identity_uuid}/edge-featurestates/all/` and `/api/v1/environments/{environment_api_key}/identities/{identity_pk}/featurestates/all/`

**Review by kyle-ssg — APPROVED** [review]
